src/gordon_gcp/plugins/publisher.py

```diff
@@ -43,14 +57,200 @@ class GDNSPublisher:
         error_channel (asyncio.Queue): a sink for
             :interface:`interfaces.IEventMessages` that were not
             processed due to problems.
+        http_client (http.AIOConnection):
+            A google http connection class
     """
     phase = 'publish'
 
-    def __init__(self, config, success_channel, error_channel):
+    def __init__(self, config, success_channel, error_channel, http_client):
         self.config = config
         self.success_channel = success_channel
         self.error_channel = error_channel
+        self.http_client = http_client
+        self.timeout = config['timeout']
+        self.managed_zone = config['managed_zone']
+        self.dns_zone = config['dns_zone']
+        self.project = config['project']
+        self.default_ttl = self.config.get('default_ttl', 300)
+        _api_version = config.get('version', 'v1')
```

Comment:
*note* I'm copying this here, since I can't respond to the earlier comment
> i'm not sure the leading '_' is required here, since it's already a local variable

> we want it to be private.. not only local..

I'm not sure I understand your meaning.  I'm not aware that private has a meaning in the context of a locally-scoped variable.

Reply:
i think she meant that we want this variable to be private so no one who has the publisher instance can use this variable, only the init function..